spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/RetrievalAugmentationAdvisor.java

```diff
@@ -67,9 +72,12 @@ public class RetrievalAugmentationAdvisor implements CallAroundAdvisor, StreamAr
 
 	private final int order;
 
-	public RetrievalAugmentationAdvisor(DocumentRetriever documentRetriever, @Nullable QueryAugmentor queryAugmentor,
-			@Nullable Boolean protectFromBlocking, @Nullable Integer order) {
+	public RetrievalAugmentationAdvisor(List<QueryTransformer> queryTransformers, DocumentRetriever documentRetriever,
+			@Nullable QueryAugmentor queryAugmentor, @Nullable Boolean protectFromBlocking, @Nullable Integer order) {
+		Assert.notNull(queryTransformers, "queryTransformers cannot be null");
+		Assert.noNullElements(queryTransformers, "queryTransformers cannot contain null elements");
 		Assert.notNull(documentRetriever, "documentRetriever cannot be null");
+		this.queryTransformers = queryTransformers;
 		this.documentRetriever = documentRetriever;
 		this.queryAugmentor = queryAugmentor != null ? queryAugmentor : ContextualQueryAugmentor.builder().build();
 		this.protectFromBlocking = protectFromBlocking != null ? protectFromBlocking : false;
```

Comment:
I think `protectFromBlocking` being true by default is a more conservative options, this way it works out of the box in a reactive environment and for a non-reactive environment there isn't much of an impact.